unified-scheduler-logic/src/lib.rs

```diff
@@ -562,91 +670,305 @@ impl Default for UsageQueueInner {
+                let mut current_and_requested_usage = (current_usage, requested_usage);
+
+                match &mut current_and_requested_usage {
+                    (None, _) => {
+                        assert!(blocked_usages_from_tasks.is_empty());
+                        Ok(())
+                    }
+                    (Some(PriorityUsage::Writable(current_task)), _) => {
+                        if !new_task.is_higher_priority(current_task) {
+                            return Err(());
+                        }
+                        if !current_task.try_reblock(token) {
+                            return Err(());
+                        }
+                        let reblocked_task = Usage::take_writable(current_and_requested_usage.0);
+                        blocked_usages_from_tasks
+                            .insert_usage_from_task((RequestedUsage::Writable, reblocked_task));
+                        Ok(())
+                    }
+                    (Some(PriorityUsage::Readonly(_current_tasks)), RequestedUsage::Readonly) => {
+                        match self.peek_usage_from_task() {
+                            None => Ok(()),
+                            Some((usage, peeked_task)) => {
+                                assert_matches!(usage, RequestedUsage::Writable);
+                                if !new_task.is_higher_priority(peeked_task) {
+                                    return Err(());
+                                }
+                                Ok(())
+                            }
+                        }
+                    }
+                    (Some(PriorityUsage::Readonly(current_tasks)), RequestedUsage::Writable) => {
+                        // Use extract_if once stablized to remove Vec creation and the repeating
+                        // remove()s...
+                        let task_indexes = current_tasks
+                            .range(new_task.task_index()..)
+                            .filter_map(|(&index, task)| task.try_reblock(token).then_some(index))
+                            .collect::<Vec<Index>>();
+                        for task_index in task_indexes.into_iter() {
+                            let reblocked_task = current_tasks.remove(&task_index).unwrap();
+                            blocked_usages_from_tasks
+                                .insert_usage_from_task((RequestedUsage::Readonly, reblocked_task));
+                        }
+
+                        if current_tasks.is_empty() {
+                            Usage::take_readable(current_and_requested_usage.0);
+                            Ok(())
+                        } else {
+                            Err(())
+                        }
+                    }
+                }
+            }
+        }
     }
 }
 
-const_assert_eq!(mem::size_of::<TokenCell<UsageQueueInner>>(), 40);
+const_assert_eq!(mem::size_of::<TokenCell<UsageQueueInner>>(), 56);
```

Comment:
this assert value changed. can you remind me what the value of asserting size is?

Reply:
it's needed to remind me of not exceeding the cache line (=64 bytes). Agreed, it's a bit of vanity metrics. on the other hand, its maintenance burden should be minimal....